crates/shared/src/sources/balancer_v2/event_handler.rs

```diff
@@ -156,93 +153,44 @@ impl BalancerPoolRegistry {
+impl<Factory> EventStoring<BasePoolFactoryEvent> for PoolStorage<Factory>
+where
+    Factory: FactoryIndexing,
+{
     async fn replace_events(
         &mut self,
-        events: Vec<EthContractEvent<WeightedPoolFactoryEvent>>,
+        events: Vec<EthContractEvent<BasePoolFactoryEvent>>,
         range: RangeInclusive<BlockNumber>,
     ) -> Result<()> {
-        self.replace_events_inner(
-            range.start().to_u64(),
-            convert_weighted_pool_created(events)?,
-        )
-        .await
-    }
+        tracing::debug!("replacing {} events for block {:?}", events.len(), range);
 
-    async fn append_events(
-        &mut self,
-        events: Vec<EthContractEvent<WeightedPoolFactoryEvent>>,
-    ) -> Result<()> {
-        tracing::debug!(
-            "inserting {} Balancer Weighted Pools from events",
-            events.len()
-        );
-        self.insert_events(convert_weighted_pool_created(events)?)
-            .await
-    }
-
-    async fn last_event_block(&self) -> Result<u64> {
-        Ok(self.last_event_block())
-    }
-}
-
-#[async_trait::async_trait]
-impl EventStoring<WeightedPool2TokensFactoryEvent> for PoolStorage {
-    async fn replace_events(
-        &mut self,
-        events: Vec<EthContractEvent<WeightedPool2TokensFactoryEvent>>,
-        range: RangeInclusive<BlockNumber>,
-    ) -> Result<()> {
-        self.replace_events_inner(
-            range.start().to_u64(),
-            convert_two_token_pool_created(events)?,
-        )
-        .await
+        self.remove_pools_newer_than_block(range.start().to_u64());
+        self.append_events(events).await
     }
 
     async fn append_events(
         &mut self,
-        events: Vec<EthContractEvent<WeightedPool2TokensFactoryEvent>>,
+        events: Vec<EthContractEvent<BasePoolFactoryEvent>>,
     ) -> Result<()> {
-        tracing::debug!(
-            "Inserting {} Balancer Weighted 2-Token Pools from events",
-            events.len()
-        );
-        self.insert_events(convert_two_token_pool_created(events)?)
-            .await
-    }
+        tracing::debug!("inserting {} events", events.len());
 
-    async fn last_event_block(&self) -> Result<u64> {
-        Ok(self.last_event_block())
-    }
-}
+        for event in events {
+            let block_created = event
+                .meta
+                .ok_or_else(|| anyhow!("event missing metadata"))?
+                .block_number;
+            let BasePoolFactoryEvent::PoolCreated(pool_created) = event.data;
 
-#[async_trait::async_trait]
-impl EventStoring<StablePoolFactoryEvent> for PoolStorage {
-    async fn replace_events(
-        &mut self,
-        events: Vec<EthContractEvent<StablePoolFactoryEvent>>,
-        range: RangeInclusive<BlockNumber>,
-    ) -> Result<()> {
-        self.replace_events_inner(range.start().to_u64(), convert_stable_pool_created(events)?)
-            .await
-    }
+            self.index_pool_creation(pool_created, block_created)
```

Comment:
I can find this method in the code when checking out locally, but I don't remember having seen a PR on it. Maybe you can add the order of PRs to the description to make it easier to navigate back and forth?

Reply:
See 👇.

I will add orders to the titles.

Reply:
Sorry, this was a bit confusing. When you reviewed the PR, I had accidentally also pushed #1463 on top of this one - so it included more changes than I wanted.